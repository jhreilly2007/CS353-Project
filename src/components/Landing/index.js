    
/**This Landing Component is created as a class-based component
because it holds all of the other states*/

import React from 'react';
import SearchBar from '../SearchBar';
import youtube from '../../apis/youtube';
import VideoList from '../VideoList';
import VideoDetail from '../VideoDetail';
import 'semantic-ui-css/semantic.min.css';
import { AuthUserContext, withAuthorization } from '../Session';



class Landing extends React.Component{
  state={
    videos: [],
    selectedVideo: null,
  }
/**assigned termFromSearchbar as we pass term from Searchber(this.prop.handleFormSubmit)
  we take the term from serachbar compoent and assign to q to that we need to pass handleSubmit 
  as a prop to the SearchBar Component(ie handleFormSubmit = this.handleSubmit)*/

    handleSubmit = async (termFromSearchBar) => {
        const response = await youtube.get('/search', {
            params: {
                q: termFromSearchBar
            }
        })
        this.setState({
            videos: response.data.items
        })
    };
    handleVideoSelect = (video) => {
        this.setState({selectedVideo: video})
    }

    render() {
        return (
            <div style={{margin: '1em'}}>
                <SearchBar handleFormSubmit={this.handleSubmit}/>
                <div style={{margin: '1em'}} className='ui grid'>
                    <div className="ui row">
                        <div className="eleven wide column">
                            <VideoDetail video={this.state.selectedVideo}/>
                        </div>
                        <div className="five wide column">
                            <VideoList handleVideoSelect={this.handleVideoSelect} videos={this.state.videos}/>
                        </div>
                    </div>
                </div>
            </div>
        )
    }
}

const condition = authUser => !!authUser;

export default Landing;
 
















/**import React, { Component } from 'react';
const API = 'https://prod.mypod.online/feed?q=';
const API_KEY = 'queen';
class Landing extends Component {
  constructor(props) {
    super(props);
    this.state = {
      hits: [],
      isLoading: false,
      error: null,
    };
  }
  componentDidMount() {
    fetch(API+API_KEY)
      .then(response => {
        if (response.ok) {
          return response.json();
        } else {
          throw new Error('Something went wrong ...');
        }
      })
      .then(result => this.setState({ hits: result.hits, isLoading: false }))
      .catch(error => this.setState({ error, isLoading: false }));
  }
  render() {
    const { hits, isLoading, error } = this.state;
    if (error) {
      return <p>{error.message}</p>;
    }
    if (isLoading) {
      return <p>Loading ...</p>;
    }
    return (
      <ul>
        {hits.map(hit =>
          <li key={hit.objectID}>
            <a href={hit.url}>{hit.title}</a>
          </li>
        )}
      </ul>
    );
  }
}
export default Landing;
*/

 
















