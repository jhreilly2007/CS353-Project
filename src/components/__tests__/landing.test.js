import React from 'react';
import { shallow, mount, render, configure } from 'enzyme';
import Landing from '../Landing';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ 
  adapter: new Adapter() 
});

  /**it('<AdminPage />', async () => {
    fetch.mockResponse(JSON.stringify(JsonData));
    const wrapper = shallow(<AdminPage />, { disableLifecycleMethods: true });
    await wrapper.instance().componentDidMount();
    wrapper.update();
    expect(fetch.mock.calls.length).toEqual(1);
    expect(wrapper.find(Table).length).toEqual(1);
  });*/

describe('LandingPage', () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<Landing debug />);
  
    expect(component).toMatchSnapshot();
  });
});

 /**it('matches the snapshot', ()=>{
  const tree =renderer.create(<AdminPage/>).toJSON();
  expect(tree).toMatchSnapshot();
})*/