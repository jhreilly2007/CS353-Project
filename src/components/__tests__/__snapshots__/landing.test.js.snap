// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingPage should render correctly in "debug" mode 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing
    debug={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <SearchBar
          handleFormSubmit={[Function]}
        />,
        <div
          className="ui grid"
          style={
            Object {
              "margin": "1em",
            }
          }
        >
          <div
            className="ui row"
          >
            <div
              className="eleven wide column"
            >
              <VideoDetail
                video={null}
              />
            </div>
            <div
              className="five wide column"
            >
              <VideoList
                handleVideoSelect={[Function]}
                videos={Array []}
              />
            </div>
          </div>
        </div>,
      ],
      "style": Object {
        "margin": "1em",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "handleFormSubmit": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="ui row"
          >
            <div
              className="eleven wide column"
            >
              <VideoDetail
                video={null}
              />
            </div>
            <div
              className="five wide column"
            >
              <VideoList
                handleVideoSelect={[Function]}
                videos={Array []}
              />
            </div>
          </div>,
          "className": "ui grid",
          "style": Object {
            "margin": "1em",
          },
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="eleven wide column"
              >
                <VideoDetail
                  video={null}
                />
              </div>,
              <div
                className="five wide column"
              >
                <VideoList
                  handleVideoSelect={[Function]}
                  videos={Array []}
                />
              </div>,
            ],
            "className": "ui row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <VideoDetail
                  video={null}
                />,
                "className": "eleven wide column",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "video": null,
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <VideoList
                  handleVideoSelect={[Function]}
                  videos={Array []}
                />,
                "className": "five wide column",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "handleVideoSelect": [Function],
                  "videos": Array [],
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
              "type": "div",
            },
          ],
          "type": "div",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <SearchBar
            handleFormSubmit={[Function]}
          />,
          <div
            className="ui grid"
            style={
              Object {
                "margin": "1em",
              }
            }
          >
            <div
              className="ui row"
            >
              <div
                className="eleven wide column"
              >
                <VideoDetail
                  video={null}
                />
              </div>
              <div
                className="five wide column"
              >
                <VideoList
                  handleVideoSelect={[Function]}
                  videos={Array []}
                />
              </div>
            </div>
          </div>,
        ],
        "style": Object {
          "margin": "1em",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "handleFormSubmit": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="ui row"
            >
              <div
                className="eleven wide column"
              >
                <VideoDetail
                  video={null}
                />
              </div>
              <div
                className="five wide column"
              >
                <VideoList
                  handleVideoSelect={[Function]}
                  videos={Array []}
                />
              </div>
            </div>,
            "className": "ui grid",
            "style": Object {
              "margin": "1em",
            },
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="eleven wide column"
                >
                  <VideoDetail
                    video={null}
                  />
                </div>,
                <div
                  className="five wide column"
                >
                  <VideoList
                    handleVideoSelect={[Function]}
                    videos={Array []}
                  />
                </div>,
              ],
              "className": "ui row",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <VideoDetail
                    video={null}
                  />,
                  "className": "eleven wide column",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "video": null,
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": <VideoList
                    handleVideoSelect={[Function]}
                    videos={Array []}
                  />,
                  "className": "five wide column",
                },
                "ref": null,
                "rendered": Object {
                  "instance": null,
                  "key": undefined,
                  "nodeType": "function",
                  "props": Object {
                    "handleVideoSelect": [Function],
                    "videos": Array [],
                  },
                  "ref": null,
                  "rendered": null,
                  "type": [Function],
                },
                "type": "div",
              },
            ],
            "type": "div",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
  Symbol(enzyme.__childContext__): null,
}
`;
